libcontainer/seccomp/seccomp_linux.go

```diff
@@ -151,17 +152,23 @@ func matchCall(filter *libseccomp.ScmpFilter, call *configs.Syscall) error {
 		return errors.New("empty string is not a valid syscall")
 	}
 
-	// If we can't resolve the syscall, assume it's not supported on this kernel
-	// Ignore it, don't error out
-	callNum, err := libseccomp.GetSyscallFromName(call.Name)
+	// Convert the call's action to the libseccomp equivalent
+	callAct, err := getAction(call.Action, call.ErrnoRet)
 	if err != nil {
+		return fmt.Errorf("action in seccomp profile is invalid: %w", err)
+	}
+	if callAct == defAct {
+		// This rule is redundant, silently skip it
+		// to avoid error from AddRule.
 		return nil
 	}
```

Comment:
We also have to compare errnoRet?

Reply:
No. The `callAct` and the `defAct` is a combo of the action (lower two bytes) and the error code (upper two bytes) -- see getAction which calls libseccomp-golang's `(ScmpAction).SetReturnCode`.